tests/plugins/test_topic_checking.py

```diff
@@ -0,0 +1,442 @@
+# Copyright (c) 2015 Nicolas JOUANIN
+#
+# See the file license.txt for copying permission.
+
+import pytest
+
+from amqtt.plugins.manager import BaseContext
+from amqtt.plugins.topic_checking import (
+    BaseTopicPlugin,
+    TopicTabooPlugin,
+    TopicAccessControlListPlugin,
+)
+from amqtt.session import Session
+
+
+class DummyLogger(object):
+    def __init__(self):
+        self.messages = []
+
+    def warning(self, *args, **kwargs):
+        self.messages.append((args, kwargs))
+
+
+# Base plug-in object
+
+
+@pytest.mark.asyncio
+async def test_base_no_config():
+    """
+    Check BaseTopicPlugin returns false if no topic-check is present.
+    """
+    context = BaseContext()
```

Comment:
I like the dummy logger approach.

Do you think it would make sense to make a pytest fixture out of the context? It would provide the base context, with the DummyLogger and an empty config.  So each test just has to change the config instead of creating the context and logger.

Reply:
That dummy logger is very minimal, to be properly useful I think I'd need to implement the full `logging.Logger` interface.  Either that, or hook a logging channel into the `logging` framework that makes the log messages available as a variable.

In fact, I think I favour the latter approach for a suite-wide dummy logger, since it'll also work for the classes where a logger was created with `logging.getLogger(__none__)`.  I'll have a look.

Reply:
I did a bit of research, just to see if there was a library that implemented some sort of "stub" `logging` module.  Looking in `pypi` was a dead loss… but I did stumble on this:

http://www.simeonvisser.com/posts/how-to-mock-logging-in-python.html

I'll have to look more closely at the `mock` module in Python to see what assert statements are available, but maybe this approach might work better for use as a dummy logger fixture?